Phase 2, SDK templates: Add configuration templates and JS example (#1812)
getsentry/sentry-docs: src/_includes/common/filter-events.md

<!--
Guideline: This page is common to all SDKs; it is stored in the common folder, nested under _includes/common. To use, 

1. Add a folder with the name of the platform you are documenting to the _documentation/sdks structure (for example, _documentation/sdks/javascript) 
2. Create a new filter.md file in _documentation/sdks/<platform-name>/config folder
3. Create the defined `include` statements and add them to the filter.md file

If you have questions, please ask Fiona or Daniel. 

**The objective for this page is that a developer can view easily the configuration options for the SDK; each page _must_ have a description that includes a summary of what the page provides to the developer. Simply linking the page is insufficient.**

> -->

Reviewer: These comments will show up in the source code, maybe we'll find a better place for them or use `{% comment %}` `{% endcomment %}` instead of `<!--` `-->`

Author: Done. Thanks.

While sending all application errors to Sentry ensures you’ll be notified in real-time when errors occur in your code, often applications generate many errors, thus many notifications. The Sentry SDKs have several configuration options you can use to filter unwanted errors from leaving your application’s runtime. In addition, the Sentry web UI also offers methods to filter events.

## Use Projects to Filter Events

Sentry provides several methods to filter data server-side. Using Sentry's web UI to filter events is a slightly simpler method as you don't have to configure and deploy your SDK.

### Inbound Data Filters

> Inbound data filters allow to determine which errors, if any, Sentry should ignore. Explore these by navigating to **[Project] » Project Settings » Inbound Filters.**

Reviewer 2: ```suggestion
Inbound data filters allow you to determine which errors, if any, Sentry should ignore. Explore these by navigating to **[Project] » Project Settings » Inbound Filters.**
```

Author: Thanks for the careful eye.

{% comment %}
Guideline: This page is common to all SDKs; it is stored in the common folder, nested under _includes/common. To use, 

1. Add a folder with the name of the platform you are documenting to the _documentation/sdks structure (for example, _documentation/sdks/javascript) 
2. Create a new filter.md file in _documentation/sdks/<platform-name>/config folder
3. Create the defined `include` statements and add them to the filter.md file

If you have questions, please ask Fiona or Daniel. 

> **The objective for this page is that a developer can view easily the configuration options for the SDK; each page _must_ have a description that includes a summary of what the page provides to the developer. Simply linking the page is insufficient.**

Author: Thanks.

## Configure your SDK to Filter Events

Configure your SDK to filter events by using the `beforeSend` callback method and configuring, enabling, or disabling integrations.

### Using `before-send`

All Sentry SDKs support the `beforeSend` callback method. `before-send` is called immediately before the event is sent to the server, so it’s the final place where you can edit its data. It receives the event object as a parameter, so you can use that to modify the event’s data or drop it completely (by returning `null`) based on custom logic and the data available on the event.

{{ include.filter-init_content }}
<!-- Guideline: add the `init` call for the SDK your are documenting -->

The `before-send` callback is passed both the `event` and a second argument, `hint`, that holds one or more hints. 

Typically a `hint` holds the original exception so that additional data can be extracted or grouping is affected. In this example, the fingerprint is forced to a common value if an exception of a certain type has been caught:

{{ include.filter-example_content }}
<!-- Guideline: add a code sample that supports this example for the SDK your are documenting -->

> Note also that breadcrumbs can be filtered, as discussed in Understand Breadcrumbs <!-- TO DO : add link-->

Reviewer 3: ```suggestion
Note also that breadcrumbs can be filtered, as discussed in Understand Breadcrumbs <!-- TODO : add link-->
```

(all programming languages use `TODO` tag instead of `TO DO`, this way most tools can find them automatically)

Author: Ah, this is an internal note to me to fix this later. But I should wrap it in the {% comment %} syntax. Thanks.

### Using Hints

Hints are available in two places: 

1. `beforeSend` / `beforeBreadcrumb` 
2. `eventProcessors` 

Event and Breadcrumb `hints` are objects containing various information used to put together an event or a breadcrumb. Typically `hints` hold the original exception so that additional data can be extracted or grouping can be affected. 

For events, those are things such as `event_id`,  `originalException`,  `syntheticException` (used internally to generate cleaner stack trace), and any other arbitrary `data` that you attach. 

For breadcrumbs, the use of `hints` is implementation dependent. For XHR requests, the hint contains the xhr object itself; for user interactions the hint contains the DOM element and event name and so forth.

In this example, the fingerprint is forced to a common value if an exception of a certain type has been caught:

{{ include.filter-hint_content }}
<!-- Guideline: add a code sample that supports the SDK your are documenting -->

#### Hints for Events

> `originalException`     The original exception that caused the Sentry SDK to create the event. This is useful for changing how the Sentry SDK groups events or to extract additional information.

Reviewer 3: Are all these whitespaces here (and below) on purpose?

Author: Between the term and its definition? Not specifically. Thanks for pointing it out (FYI -  I have planned time at the end of the final phase to complete format fixes that pop up, but am glad you pointed this out here).
